### PR TITLE
Add markdown link CI check

### PR DESCRIPTION
**Pull request by lalitb**
## Changes

Shamelessly copied from https://github.com/open-telemetry/opentelemetry-specification/blob/5d9cef817415a15aaa4883e184a38adec79096ed/.github/workflows/checks.yaml#L36, and fixed broken links.

This will recursively check all the markdown files (ignoring CHANGELOG.md), and fail if any dead link is found.

If a CI check isn't necessary, I can modify the PR to exclude it, focusing solely on the updates to the broken links.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: scripts/markdown-link-check-config.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+"httpHeaders": [
+    {
+      "urls": ["https://crates.io"],
+      "headers": {
+        "Accept": "text/html"
```
Comment (lalitb) [inline scripts/markdown-link-check-config.json:6]: HTTP request on crates.io fails with a `404` error, if the `Accept: text/html` header is not specified.

File: opentelemetry-sdk/README.md [hunk 2]
```diff
@@ -34,3 +34,8 @@ From the root directory, run the following command:
 ```sh
 cargo bench
 ```
+
+## Supported Rust Versions
```
Comment (cijothomas) [inline opentelemetry-sdk/README.md:38]: we need to put it in main readme, and everywhere else can just point to it

File: .github/workflows/markdown-link-check.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: Markdown link check
+
+on:
+  pull_request:
```
Comment (cijothomas) [inline .github/workflows/markdown-link-check.yml:4]: maybe once a week is fine?

File: opentelemetry-sdk/README.md (hunk 2 above)
Comment (lalitb) [reply opentelemetry-sdk/README.md:38]: Yeah I thought so, but then kept it consistent with other crates from this repo. 

https://github.com/search?q=repo%3Aopen-telemetry%2Fopentelemetry-rust+%22Supported+Rust+Versions%22&type=code

File: .github/workflows/markdown-link-check.yml (hunk 3 above)
Comment (lalitb) [reply .github/workflows/markdown-link-check.yml:4]: As of now it only runs for PRs having markdown changes and takes ~40 secs to complete. But no strong opinion, once a week works fine too if we all agree :)

Comment (TommyCpp) [reply .github/workflows/markdown-link-check.yml:4]: if it's ~40 sec maybe we can just run it for every PR that touches markdown?

**Review by cijothomas — APPROVED** [review]

**Pull request merged**